### ISSUE TITLE
Deep Fluids: A Generative Network for Parameterized Fluid Simulations

### ISSUE THREAD

**Issue by j20232** [j20232/survey#17]
![image](https://user-images.githubusercontent.com/13940471/69349367-f7db0d80-0cba-11ea-8a9c-6c2be3a305fc.png)

## Summary
- CNN to synthesize plausible and fully divergence-free 2-D and 3-D fluid simulation velocities
- Generative model for fluids that accurately encodes parameterizable velocity fields
- Approach to encode simulation classes into a latent space representation through AE

## Abstract

This paper presents a novel generative model to synthesize fluid simulations from a set of reduced parameters. A convolutional neural network is trained on a collection of discrete, parameterizable fluid simulation velocity fields. Due to the capability of deep learning architectures to learn representative features of the data, our generative model is able to accurately approximate the training data set, while providing plausible interpolated in-betweens. The proposed generative model is optimized for fluids by a novel loss function that guarantees divergence-free velocity fields at all times. In addition, we demonstrate that we can handle complex parameterizations in reduced spaces, and advance simulations in time by integrating in the latent space with a second network. Our method models a wide variety of fluid behaviors, thus enabling applications such as fast construction of simulations, interpolation of fluids with different parameters, time re-sampling, latent space simulations, and compression of fluid simulation data. Reconstructed velocity fields are generated up to 700x faster than re-simulating the data with the underlying CPU solver, while achieving compression rates of up to 1300x.

## Author

- <a href="https://arxiv.org/search/cs?searchtype=author&amp;query=Kim%2C+B">Byungsoo Kim</a>
- <a href="https://arxiv.org/search/cs?searchtype=author&amp;query=Azevedo%2C+V+C">Vinicius C. Azevedo</a>
- <a href="https://arxiv.org/search/cs?searchtype=author&amp;query=Thuerey%2C+N">Nils Thuerey</a>
- <a href="https://arxiv.org/search/cs?searchtype=author&amp;query=Kim%2C+T">Theodore Kim</a>
- <a href="https://arxiv.org/search/cs?searchtype=author&amp;query=Gross%2C+M">Markus Gross</a>
- <a href="https://arxiv.org/search/cs?searchtype=author&amp;query=Solenthaler%2C+B">Barbara Solenthaler</a>

## Journal/Conference

Computer Graphics Forum (Proc. Eurographics), 38, 2 (2019), 59-70
## Subjects

- `cs.LG`: Machine Learning
- `cs.GR`: Graphics
- `physics.comp-ph`: Computational Physics
- `physics.flu-dyn`: Fluid Dynamics
- `stat.ML`: Machine Learning

## Comment

Computer Graphics Forum (Proceedings of EUROGRAPHICS 2019), additional materials: <a class="link-external link-http" href="http://www.byungsoo.me/project/deep-fluids/" rel="external noopener nofollow">this http URL</a>

## Link

- arXiv: https://arxiv.org/abs/1806.02071
- Official: http://www.byungsoo.me/project/deep-fluids/
- YouTube: https://youtu.be/hSDzOZ9IO8U
- Code: https://github.com/byungsook/deep-fluids
- Seamless: https://shiropen.com/seamless/deep-fluids
